Update tests for compatibility with Magento v2.4-develop

fredden opened the pull request:
Unit & integration tests are currently failing with recent changes in `2.4-develop` branch. This pull request resolves the unit test failure.

The failure for integration tests relates to Magento v2.4 requiring elasticsearch to be installed and available, but I have not looked into how to set this up in Travis CI yet.

Magento v2.4 will not support PHP v7.2, so these tests have been removed too.

fredden committed:
Update unit test for compatibility with v2.4

fredden committed:
No longer test Magento v2.4 with PHP v7.2

Magento v2.4 will not support PHP v7.2, so these tests will never pass.

Ethan3600 commented:
Might even be worth commenting out the 2.4 integration for now until we can fully support it. What do you think?

Outside of that, I think if we can fix the non-2.4 related integration tests, this is good to go!

Looks like `EthanYehuda\CronjobManager\Test\Integration\ProcessIdTest::testProcessIdSavedOnStart` is failing.

fredden committed:
Enable elasticsearch service for Travis CI

fredden committed:
Set return types for setUp() & tearDown() methods

fredden committed:
Merge remote-tracking branch 'upstream/1.x' into fix-tests-for-2.4-develop

fredden committed:
Update integration tests for compatibility w/ v2.4

fredden committed:
Add unit test for runjob console command

fredden committed:
Add PHP v7.4 support and enable tests

fredden committed:
Declare dependency on posix extension

fredden committed:
Support PHPUnit before & after assertion change

fredden committed:
Remove tests with PHP v7.4 temporarily

Travis CI's build of PHP v7.4 does not include the 'zip' nor 'gd' extensions.
These are required as part of core Magento and therefore these tests fail.

fredden commented:
Tests are now all green, but I don't know how long that'll last. Magento v2.4-develop seems to be a moving target lately.

Ethan3600 commented:
Sweet! Awesome work! Going to merge

Ethan3600 commented:
Let's see how it plays out with Magento 2.4, I wouldn't be upset if we had to comment those out if it gets crazy.
